src/main/java/de/tum/in/www1/artemis/repository/ModelingExerciseRepository.java

```diff
@@ -97,6 +97,24 @@ default ModelingExercise findByIdElseThrow(long exerciseId) {
         return findById(exerciseId).orElseThrow(() -> new EntityNotFoundException("Modeling Exercise", exerciseId));
     }
 
+    /**
+     * Find a modeling exercise by its id, with grading criteria loaded, and throw an EntityNotFoundException if it cannot be found
+     *
+     * @param exerciseId of the modeling exercise.
+     * @return The modeling exercise related to the given id
+     */
+    @Query("""
+            SELECT DISTINCT e
+            FROM ModelingExercise e
+                LEFT JOIN FETCH e.gradingCriteria
+            WHERE e.id = :exerciseId
+            """)
+    Optional<ModelingExercise> findByIdWithGradingCriteria(@Param("exerciseId") long exerciseId);
+
+    default ModelingExercise findByIdWithGradingCriteriaElseThrow(long exerciseId) {
```

Comment:
shouldn't this method be tagged with `@NotNull`, since it will throw if the exercise isn't found?